### PR TITLE
Vb 3869 refactor booker dto and associated entities to have a more re…

### PR DESCRIPTION
**Pull request by AledWEvans**
Refactor booker dto and associated entities to have a more relevant naming, new pr as had major merge issues.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/BookerDetailsService.kt [hunk 1]
```diff
@@ -43,35 +41,67 @@ class BookerDetailsService(
   @Transactional
   fun clearBookerDetails(bookerReference: String): BookerDto {
     val booker = getBooker(bookerReference)
-    booker.prisoners.clear()
+    booker.permittedPrisoners.clear()
     return BookerDto(bookerRepository.saveAndFlush(booker))
   }
 
+  @Transactional(readOnly = true)
+  fun getAssociatedPrisoners(reference: String, active: Boolean?): List<PermittedPrisonerDto> {
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/BookerDetailsService.kt:49]: minor - does this method need to be getPermittedPrisoners now?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/BookerDetailsService.kt [hunk 2]
```diff
@@ -43,35 +41,67 @@ class BookerDetailsService(
...
+    val bookerByReference = getBooker(reference)
+    val associatedPrisoners =
+      active?.let {
+        prisonerRepository.findByBookerIdAndActive(bookerByReference.id, it)
+      } ?: prisonerRepository.findByBookerId(bookerByReference.id)
+    return associatedPrisoners.map(::PermittedPrisonerDto)
+  }
+
+  fun getAssociatedPrisoner(reference: String, prisonerId: String): PermittedPrisoner? {
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/BookerDetailsService.kt:58]: same as above

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/BookerDetailsService.kt [hunk 3]
```diff
@@ -43,35 +41,67 @@ class BookerDetailsService(
...
+    val bookerByReference = getBooker(reference)
+    return prisonerRepository.findByBookerIdAndPrisonerId(bookerByReference.id, prisonerId)
+  }
+
+  @Transactional(readOnly = true)
+  fun getAssociatedVisitors(bookerReference: String, prisonerId: String, active: Boolean?): List<PermittedVisitorDto> {
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/BookerDetailsService.kt:64]: same as above

**Review by dhirajshettymoj — APPROVED** [review]
LGTM - minor comment added around method names. Can be ignored.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/integration/PrisonersVisitorsByBookerReferenceTest.kt [hunk 4]
```diff
@@ -157,23 +142,39 @@ class PrisonersVisitorsByBookerReferenceTest : IntegrationTestBase() {
...
+    return objectMapper.readValue(returnResult.returnResult().responseBody, Array<PermittedVisitorDto>::class.java).toList()
   }
 
-  private fun assertVisitorDetails(visitor: VisitorDto, visitorDetails: PrisonersVisitorDetails) {
+  private fun assertVisitorDetails(visitor: PermittedVisitorDto, visitorDetails: PermittedVisitorDetails) {
     Assertions.assertThat(visitor.visitorId).isEqualTo(visitorDetails.visitorId)
     Assertions.assertThat(visitor.active).isEqualTo(visitorDetails.isActive)
   }
+
+  fun getPrisonerVisitorsByBookerReference(
```
Comment (adaviesMOJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/integration/PrisonersVisitorsByBookerReferenceTest.kt:160]: getPermittedPrisonerVisitorsByBookerReference to keep with the theme?

**Review by adaviesMOJ — APPROVED** [review]
LGTM, 2 very minor suggestions

**Review by adaviesMOJ — DISMISSED** [review]

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Review by adaviesMOJ — APPROVED** [review]

**Pull request merged**